### DISCUSSION TITLE
Nav events not firing?

### DISCUSSION THREAD

**Discussion by byron-roots** [statamic/cms#4355] [Troubleshooting] [answered]
I've had to employ some quite extensive caching on my site to keep the page speed to an acceptable level. I've then used event listeners to clear the certain cache tags to so that newly edited content is shown.

In my event service provider I have the following in the boot function
```
    /**
     * Register any events for your application.
     *
     * @return void
     */
    public function boot()
    {
        Event::listen(function (EntrySaved $event) {
            Cache::tags($event->entry->id())->flush();
            Cache::tags($event->entry->collectionHandle() . '_collection')->flush();
        });

        Event::listen(function (NavSaved $event) {
            Cache::tags($event->nav->id())->flush();
            Cache::tags($event->nav->handle() . '_navigation')->flush();
        });
    }
```

The `EntrySaved` listener is working fine and the cache is being flushed as expected. The `NavSaved` listener is not being called when I save a nav. I've tried dropping a `Log::info(event->nav->handle() . ' Saved')` and nothing turns up in the logs. Have I configured this wrong, or is something else happening?

**Comment by jasonvarga** [comment, marked as answer]
There's a `NavSaved` event for when the Nav itself is saved. That's the file that holds the configuration. The page in the CP where you edit the title, if you save that, you'll see that event.

When you're moving items around in the tree, that'll fire the `NavTreeSaved` event. Use that!

**Reply by byron-roots** [reply]
Excellent that's fixed it, thought I was missing something!
